src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CreateServer.cs

```diff
@@ -0,0 +1,219 @@
+        [InlineData(PipeDirection.Out)]
+        public static void BadHandleKind_Throws_IOException(PipeDirection direction)
+        {
+            using (FileStream fs = new FileStream(Guid.NewGuid().ToString(), FileMode.Create, FileAccess.Write, FileShare.None, 8, FileOptions.DeleteOnClose))
+            {
+                SafeFileHandle safeHandle = fs.SafeFileHandle;
+
+                bool gotRef = false;
+                try
+                {
+                    safeHandle.DangerousAddRef(ref gotRef);
+                    IntPtr handle = safeHandle.DangerousGetHandle();
+
+                    SafePipeHandle fakePipeHandle = new SafePipeHandle(handle, ownsHandle: false);
+                    Assert.Throws<IOException>(() => new NamedPipeServerStream(direction, false, true, fakePipeHandle));
+                }
+                finally
+                {
+                    if (gotRef)
+                        safeHandle.DangerousRelease();
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        [PlatformSpecific(PlatformID.Windows)]
+        public static void Windows_CreateFromDisposedServerHandle_Throws_ObjectDisposedException(PipeDirection direction)
+        {
+            // The pipe is closed when we try to make a new Stream with it
+            var pipe = new NamedPipeServerStream(GetUniquePipeName(), direction, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous);
+            var handle = pipe.SafePipeHandle;
+            pipe.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => new NamedPipeServerStream(direction, true, true, pipe.SafePipeHandle).Dispose());
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public static void Unix_GetHandleOfNewServerStream_Throws_InvalidOperationException()
+        {
+            var pipe = new NamedPipeServerStream(GetUniquePipeName(), PipeDirection.Out, 1, PipeTransmissionMode.Byte);
+            Assert.Throws<InvalidOperationException>(() => pipe.SafePipeHandle);
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        [PlatformSpecific(PlatformID.Windows)]
+        public static void Windows_CreateFromAlreadyBoundHandle_Throws_ArgumentException(PipeDirection direction)
+        {
+            // The pipe is already bound
+            var pipe = new NamedPipeServerStream(GetUniquePipeName(), direction, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous);
+            Assert.Throws<ArgumentException>(() => new NamedPipeServerStream(direction, true, true, pipe.SafePipeHandle));
+            pipe.Dispose();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // NumberOfServerInstances > 1 isn't supported and has undefined behavior on Unix
```

Comment:
Nope. Oddly enough, we do the same validity checks on it that we do for Windows: check that it is between 1 and 255.